### ISSUE TITLE
Torch, torcaudio etc. won't pip install from inside pyenv-virtualenv

### ISSUE THREAD

**Issue by IdoWSC** [pytorch/pytorch#53601]
## 🐛 Bug

When trying to install torch and other affiliates in a pyenv-virtualenv virtual environment I get the error:

```
ERROR: Could not find a version that satisfies the requirement torchaudio==0.5.1 (from -r requirements.txt (line 8)) (from versions: none)
ERROR: No matching distribution found for torchaudio==0.5.1 (from -r requirements.txt (line 8))

```

Also when downloading the whl file from https://download.pytorch.org/whl/torch_stable.html and trying to explicitly install it by
`pip install torch-1.5.1-cp37-none-macosx_10_9_x86_64.whl`

I get
`ERROR: torch-1.5.1-cp37-none-macosx_10_9_x86_64.whl is not a supported wheel on this platform.`

When deactivating the virtualenv both work perfectly. 

## To Reproduce

1. pyenv virtualenv 3.7.9 denoiser
2. pyenv activate denoiser
3. pip install -r requirements.txt 

ERROR: Could not find a version that satisfies the requirement torchaudio==0.5.1 (from -r requirements.txt (line 8)) (from versions: none)
ERROR: No matching distribution found for torchaudio==0.5.1 (from -r requirements.txt (line 8))

ERROR: Could not find a version that satisfies the requirement torch==1.5.1 (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch==1.5.1

## Expected behavior

Torch will install like outside the virtualenv

## Environment

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: macOS 11.1 (x86_64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.28)
CMake version: Could not collect

Python version: 3.7 (64-bit runtime)
Is CUDA available: N/A
CUDA runtime version: Could not collect
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect

**Comment by mrshenli** [comment]
cc @vincentqb looks like this issue belongs to torchaudio repo?

**Comment by IdoWSC** [comment]
@mrshenli it happens also with 
`pip install torch`

**Comment by IdoWSC** [comment]
anyone?

**Comment by vincentqb** [comment]
> Also when downloading the whl file from https://download.pytorch.org/whl/torch_stable.html and trying to explicitly install it by
> `pip install torch-1.5.1-cp37-none-macosx_10_9_x86_64.whl`
> 
> I get
> `ERROR: torch-1.5.1-cp37-none-macosx_10_9_x86_64.whl is not a supported wheel on this platform.`
> 
> When deactivating the virtualenv both work perfectly.

I can indeed install torchaudio==0.5.1 inside a conda environment
```
❯ conda create -n audio-old python==3.7
audio-old ❯ conda activate audio-old
audio-old ❯ pip install torchaudio==0.5.1
```
This seem to be specific to pyenv and torch then? @seemethere

**Comment by rgommers** [comment]
This is very likely to be a a case of "`pip` is too old". OS version reported here is: `OS: macOS 11.1 (x86_64)`. The wheels are named `macosx_10_9_x86_64.whl`. Older versions of `pip` will fail to pick up `10_9` wheels, this was only fixed in Pip `21.2` or `21.3` IIRC. 

I'm quite certain that will fix the problems you were seeing, so I will close this issue. Please comment/reopen if you still see this with pip `21.3.1` or higher.

Also note that you can, as a workaround, make macOS report its version as `10.16` even if it is `11.x` with: `export SYSTEM_VERSION_COMPAT=1`. That way `pip` will pick up the `10_9` wheels.

> This seems to be specific to pyenv and torch then?

It's not specific to `pyenv`, you should see this even if you run `pip install torch` with an older `pip` inside a `conda` environment, or with system Python.

**Comment by rgommers** [comment]
Also xref gh-47978, this is a duplicate from that issue.
